jsonrpc/client.go

```diff
@@ -53,37 +53,75 @@ func (sc *Client) AccountNonce(context.Context, string) (*big.Int, error) {
 	panic("not implemented")
 }
 
+func (sc *Client) BlockNumber(ctx context.Context) (*big.Int, error) {
+	var blockNumber big.Int
+	err := sc.c.CallContext(ctx, &blockNumber, "starknet_blockNumber")
+	if err != nil {
+		return nil, err
+	}
+	return &blockNumber, err
+}
+
+func (sc *Client) LatestBlock(ctx context.Context, scope string) (*types.Block, error) {
+	blockNumber, err := sc.BlockNumber(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := sc.BlockByNumber(ctx, blockNumber, scope)
+	if err != nil {
+		return nil, err
+	}
+
+	return block, err
+}
+
 func (sc *Client) BlockByHash(ctx context.Context, hash string, scope string) (*types.Block, error) {
-	return sc.getBlock(ctx, "starknet_getBlockByHash", hash, scope)
+	var block types.Block
+	err := sc.fetchStructure(ctx, "starknet_getBlockByHash", block, hash, scope)
+
+	return &block, err
 }
 
 func (sc *Client) BlockByNumber(ctx context.Context, number *big.Int, scope string) (*types.Block, error) {
-	return sc.getBlock(ctx, "starknet_getBlockByNumber", toBlockNumArg(number), scope)
+	var block types.Block
+	err := sc.fetchStructure(ctx, "starknet_getBlockByNumber", block, toBlockNumArg(number), scope)
+
+	return &block, err
```

Comment:
fix: pass block by reference so unmarshal
fix: return nil on error, it makes the code more explicit and easier to understand
nit: minimize err scope


```suggestion
	if err := sc.fetchStructure(ctx, "starknet_getBlockByNumber", &block, toBlockNumArg(number), scope); err != nil {
		return nil, err
	}

	return &block, nil
```